infra/deploy_kubernetes_manifests.tf

```diff
@@ -94,6 +94,7 @@ resource "kubernetes_job" "kubernetes_manifests_deployer_job" {
     google_project_iam_member.google_service_account_is_kubernetes_admin,
     google_service_account_iam_member.allow_kubernetes_sa_to_impersonate_google_cloud_sa,
     kubernetes_service_account.kubernetes_manifests_deployer_service_account,
+    module.enable_multi_cluster_google_apis,
```

Comment:
We're adding this `depends_on` because this Kubernetes Job resource deploys MCI resources (i.e., should wait for `module.enable_multi_cluster_google_apis`).
Prior to this pull-request, this explicit `depends_on` `module.enable_multi_cluster_google_apis` wasn't necessary because the `depends_on` `google_container_cluster.my_cluster_europe` &
`google_container_cluster.my_cluster_usa` guaranteed multi-cluster APIs were enabled.